dataactvalidator/config/sqlRules.csv

```diff
@@ -1,7 +1,14 @@
 "SELECT af.row_number, af.bydirectreimbursablefundingsource FROM award_financial AS af WHERE af.submission_id = {} AND COALESCE(lower(af.bydirectreimbursablefundingsource),'') NOT IN ('', 'r', 'd')",B12,"Reimbursable flag indicator is required when reporting obligation or outlay USSGL account balances (excluding downward adjustments SGL accounts). Valid values are ""R"" for ""ReimbursableFundingSource"" and ""D"" for ""DirectFundingSource"".","The reimbursable flag indicator (bydirectreimbursablefundingsource) must be empty, R, or D",n,award_financial,fatal
 "SELECT op.row_number, op.bydirectreimbursablefundingsource FROM object_class_program_activity AS op WHERE op.submission_id = {} AND COALESCE(lower(op.bydirectreimbursablefundingsource),'') NOT IN ('', 'r', 'd')",B12,"Reimbursable flag indicator is required when reporting obligation or outlay USSGL account balances (excluding downward adjustments SGL accounts). Valid values are ""R"" for ""ReimbursableFundingSource"" and ""D"" for ""DirectFundingSource"".","The reimbursable flag indicator (bydirectreimbursablefundingsource) must be empty, R, or D",n,program_activity,fatal
+"SELECT af.row_number, af.objectclass from award_financial AS af WHERE af.submission_id = {} AND af.objectclass NOT IN (SELECT object_class_code FROM object_class)",B11,"Must be valid 3-digit object class as defined in OMB Circular A-11 Section 83.6, or a 4-digit code which includes a 1-digit prefix that distinguishes direct, reimbursable, and allocation obligations. Do not include decimal points when reporting in the Schema.","Failed rule: Must be valid 3-digit object class as defined in OMB Circular A-11 Section 83.6, or a 4-digit code which includes a 1-digit prefix that distinguishes direct, reimbursable, and allocation obligations. Do not include decimal points when reporting in the Schema.",n,award_financial,warning
+"SELECT af.row_number, af.objectclass, af.bydirectreimbursablefundingsource FROM award_financial af WHERE af.submission_id = {} AND LENGTH(af.objectclass) = 4 AND ((LEFT(af.objectclass,1) = '1' AND lower(bydirectreimbursablefundingsource) IS DISTINCT FROM 'd') OR (LEFT(af.objectclass,1) = '2' AND lower(af.bydirectreimbursablefundingsource) IS DISTINCT FROM 'r'))",B11,"Must be valid 3-digit object class as defined in OMB Circular A-11 Section 83.6, or a 4-digit code which includes a 1-digit prefix that distinguishes direct, reimbursable, and allocation obligations. Do not include decimal points when reporting in the Schema.","Failed rule: If object class is 4 digits, check that 1 digit prefix is consistent with value for ByDirectReimbursableFundingSource",n,award_financial,warning
```

Comment:
Yep, thanks for catching that--it was a cut-and-paste-merge-conflict-resolution gone awry. Also, I had a commit somewhere that removed the existing B11. Will find out what happened to that and put it back.

Also, thank you very much for digging into the unittest differences. Re: the prefix question, the rule as written assumes that 4 digit objectclasses with a prefix other than 1 or 2 will generate errors in the SQL rule that validates against the objectclass domain value list. That assumption holds true for the current set of domain values (the only 4 digit objectclasses on that list begin w/ 1 and 2).

So the question is...if we get an `ABCD` objectclass, is it enough to return an invalid object class message, or do we also need to tell users about an invalid prefix?